File: abc323a/abc323a.ml
open Printf
open Scanf

let solve s =
  let rec loop i =
    if i > 15 then "Yes"
    else if s.[i] = '0' then loop (i + 2)
    else "No" in
  loop 1

let () =
  scanf "%s " solve |> printf "%s\n"

File: abc323a/abc323a.hs
import Data.Bool (bool)

main :: IO ()
main = solve <$> getLine >>= putStrLn

solve :: String -> String
solve = bool "No" "Yes" . all (== '0') . f []
  where
    f ls [] = ls
    f ls (_ : a : as) = f (a : ls) as
